File: src/main/java/Chap2/Observable/ObservableFromCallable.java
package Chap2.Observable;

import io.reactivex.rxjava3.core.Observable;

import java.util.concurrent.Callable;

public class ObservableFromCallable {
    private void emit() {
        Callable<String> callable = () -> {
            Thread.sleep(1000);
            return "Hello Callable";
        };
        Observable<String> source = Observable.fromCallable(callable);
        source.subscribe(System.out::println);
    }


    public static void main(String[] args) {
        ObservableFromCallable demo=new ObservableFromCallable();
        demo.emit();
    }
}

File: src/main/kotlin/Chap2/ObservableFromCallable.kt
package Chap2

import io.reactivex.rxjava3.core.Observable
import java.util.concurrent.Callable

class ObservableFromCallable {
    fun emit() {
        val callable = Callable {
            Thread.sleep(1000)
            "Hello Callable"

        }
        val source= Observable.fromCallable(callable)
        source.subscribe(System.out::println)
    }
}

fun main() {
    val demo=ObservableFromCallable()
    demo.emit()
}